### PR TITLE
Budget App Feature branch Pull Request

### PR DESCRIPTION
**Pull request by AleWaweru**
In this project the following was implemented

- [ ] Splash screen

- A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- [ ] Sign up and log in pages

- The user should be able to register in the app with full name, email and password (all mandatory).

- The user can log into the app using email and password.

- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] Home page (categories page)

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [ ] Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new transaction" page

- The user fills out a form to create a new transaction with:
- name (mandatory)
- amount (mandatory)
- categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,189 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+I would like to thank Microverse for giving me this opportunity to build this amazing project. 
+I would also like to thank **Gregoire Vella** for the wonderful design template. I would also like to thank my mentors for their guidance and support. Lastly, I would like to thank the open-source community for their contribution to the open-source ecosystem. Without their contribution, creating this project would not have been possible. Thank you all. 🙏
+
+
```
Comment (fobadara) [inline README.md]: [REQUIRED] Kindly give appropriate credit to Gregoire Vella according to the Creative Commons license. You are missing a link to the author. Kindly check this [link](https://creativecommons.org/licenses/by-nc/4.0/) to know what is requirement for the acknowledgement.

File: app/views/static_pages/splash.html.erb
Comment (fobadara) [inline app/views/static_pages/splash.html.erb:1]: [REQUIRED] I noticed that you have a splash screen but I can't access the page unless I am logged in and it defeats the purpose of a splash screen. Kindly make it accessible and the first page shows when a user is not logged in.

[REQUIRED] Also kindly ensure that your design matches the template. Your login and signup buttons should be at the lower part of the screen and your logo in the center:

image

File: app/views/categories/index.html.erb
Comment (fobadara) [inline app/views/categories/index.html.erb:1]: [REQUIRED] Kindly ensure that the design for your categories page looks like the template. In your  design the image on the left (in this image that is the house and car) should fill the entire left part of the screen and your texts should be on the right

Your design
image

Template
image

File: app/views/transactions/index.html.erb
Comment (fobadara) [inline app/views/transactions/index.html.erb:1]: [REQUIRED] Your transactions page should have a total amount of transactions at the top of the page. Kindly implement this.

[REQUIRED] Please ensure that your design matches the template in the image below:

image

File: app/views/transactions/new.html.erb
Comment (fobadara) [inline app/views/transactions/new.html.erb:1]: [REQUIRED] When creating a new transaction a user must enter the name, amount and at least one category but in your app a user can submit an empty transaction. Kindly fix this

File: app/views/categories/index.html.erb
Comment (fobadara) [inline app/views/categories/index.html.erb:1]: [REQUIRED] The top part of your homepage doesnt match the design. Kindly fix this. 
Your design:
image

Template 
image

File: config/routes.rb
Comment (fobadara) [inline config/routes.rb:1]: [REQUIRED]  There should be a root page in your app and this should be the splash page. Kindly implement this.

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @AleWaweru, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

REDO

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Good first attempt
✅ Good implementation of the principle of separating concerns.

## Required Changes ♻️
_Check the comments under the review._  
[REQUIRED] Kindly ensure that your background color and text colors match the template. I noticed that you have not followed the template properly throughout your app.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- I think you should make use of describe blocks in your specs

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (AleWaweru) [reply README.md]: done.

File: app/views/categories/index.html.erb
Comment (AleWaweru) [reply app/views/categories/index.html.erb:1]: done

File: app/views/transactions/index.html.erb
Comment (AleWaweru) [reply app/views/transactions/index.html.erb:1]: done

File: app/views/transactions/new.html.erb
Comment (AleWaweru) [reply app/views/transactions/new.html.erb:1]: done.

File: app/views/categories/index.html.erb
Comment (AleWaweru) [reply app/views/categories/index.html.erb:1]: done

File: app/views/static_pages/splash.html.erb
Comment (AleWaweru) [reply app/views/static_pages/splash.html.erb:1]: done.

File: config/routes.rb
Comment (AleWaweru) [reply config/routes.rb:1]: done

**Review by binhussen — APPROVED** [review]
Hi @AleWaweru , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Sticked Given Design ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**